app/controllers/api/v1/packages_controller.rb

```diff
@@ -88,17 +94,54 @@ def add_item_to_stockit
     def package_params
       attributes = [:quantity, :length, :width, :height, :notes, :item_id,
         :received_at, :rejected_at, :package_type_id, :state_event, :image_id,
-        :inventory_number, :location_id]
+        :inventory_number, :location_id, :designation_name]
       params.require(:package).permit(attributes)
     end
 
+    def item_attributes
+      if (item = item_params)
+        item["donor_condition_id"] = DonorCondition.find_by(name_en: item["donor_condition_id"]).try(:id)
+        item
+      end
+    end
+
+    def item_params
+      params["package"].require(:item).permit(:donor_condition_id)
+    end
+
+    def save_item_details
+      params["package"]["item"] &&
+      (attributes = item_attributes) &&
+      @package.item.update_attributes(attributes)
+    end
+
     def serializer
       Api::V1::PackageSerializer
     end
 
     def offer_id
-      Item.where(id: @package.item_id).pluck(:offer_id).first
+      @package.item.offer_id
     end
 
+    def package_record
+      if package_params[:inventory_number]
+        if existing_package
```

Comment:
can you make this `Package.find_by(inventory_number: package_params[:inventory_number])` and remove `existing_package` as it makes the logic easier to follow. I wasn't sure what `existing_package` means, otherwise we can rename it from `existing_package` to `find_package_by_inventory_number` but I think it only saves us one line.